ProcessLib/BoundaryCondition/BHEBottomDirichletBoundaryCondition.h

```diff
@@ -0,0 +1,115 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "BoundaryCondition.h"
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+#include "NumLib/IndexValueVector.h"
+#include "ProcessLib/HeatTransportBHE/BHE/BHEAbstract.h"
+#include "ProcessLib/Parameter/Parameter.h"
+
+namespace ProcessLib
+{
+class BHEBottomDirichletBoundaryCondition final : public BoundaryCondition
+{
+public:
+    BHEBottomDirichletBoundaryCondition(
+        GlobalIndexType global_idx_T_in_bottom,
+        GlobalIndexType global_idx_T_out_bottom,
+        MeshLib::Mesh const& bulk_mesh,
+        std::vector<MeshLib::Node*> const& vec_outflow_bc_nodes,
+        int const variable_id,
+        unsigned const integration_order,
+        std::size_t const bulk_mesh_id,
+        int const component_id,
+        unsigned const bhe_idx)
+        : _bulk_mesh_id(bulk_mesh_id), _bulk_mesh(bulk_mesh)
+    {
+        DBUG(
+            "Found %d nodes for BHE bottom Dirichlet BCs for the variable %d "
+            "and "
+            "component %d",
+            vec_outflow_bc_nodes.size(), variable_id, component_id);
+
+        MeshLib::MeshSubset bc_mesh_subset{_bulk_mesh, vec_outflow_bc_nodes};
+
+        // create memory to store Tout values
+        _T_in_values.ids.clear();
+        _T_in_values.values.clear();
+
+        SpatialPosition pos;
+
+        _bc_values.ids.clear();
+        _bc_values.values.clear();
+
+        // convert mesh node ids to global index for the given component
+        _bc_values.ids.reserve(bc_mesh_subset.getNumberOfNodes());
+        _bc_values.values.reserve(bc_mesh_subset.getNumberOfNodes());
+        for (auto const* const node : bc_mesh_subset.getNodes())
+        {
+            pos.setNodeID(node->getID());
+            MeshLib::Location l(_bulk_mesh_id, MeshLib::MeshItemType::Node,
+                                node->getID());
+            // that might be slow, but only done once
+            const auto g_T_out_idx = global_idx_T_out_bottom;
```

Comment:
minor; I'd not introduce a new variable just to save only half of the characters.

Reply:
removed.